src/optimizer/join_order/cardinality_estimator.cpp

```diff
@@ -564,13 +564,13 @@ void CardinalityEstimator::EstimateBaseTableCardinality(JoinNode *node, LogicalO
 	auto table_filters = GetTableFilters(op);
 
 	auto card_after_filters = node->GetBaseTableCardinality();
-	// Logical Filter on a seq scan
-	if (has_logical_filter) {
-		card_after_filters *= DEFAULT_SELECTIVITY;
-	} else if (table_filters) {
+	if (table_filters) {
 		double inspect_result = (double)InspectTableFilters(card_after_filters, op, table_filters);
 		card_after_filters = MinValue(inspect_result, (double)card_after_filters);
 	}
+	if (has_logical_filter) {
+		card_after_filters *= DEFAULT_SELECTIVITY;
```

Comment:
This is indeed the code that caused the bug. It checked whether there was a `LogicalFilter`. If there is one, it would ignore the filters that were pushed into the base tables. We now apply multiple filter selectivities independently, but I don't see why that would be a problem.

In `InspectTableFilters`, we only apply the filters on the base table, and not those in the `LogicalFilter` that might be on top. So the bug is similar. Maybe we can check together how to properly fix this tomorrow?